src/servers/http_server_v2.cc

```diff
@@ -1177,28 +1217,110 @@ HTTPAPIServerV2::EVBufferToInput(
       // FIXMEV2 handle non-raw content types
       char* base;
       size_t byte_size;
-      RETURN_IF_ERR(ReadDataFromJson(request_input, &base, &byte_size));
+      rapidjson::Value::ConstMemberIterator itr =
+          request_input.FindMember("parameters");
+      if (itr != request_input.MemberEnd()) {
+        if (header_length == 0) {
+          return TRTSERVER_ErrorNew(
+              TRTSERVER_ERROR_INVALID_ARG,
+              "must specify valid 'Infer-Header-Content-Length' in request "
+              "header");
+        }
+        // Handle binary data case
+        const rapidjson::Value& params = itr->value;
+        size_t binary_offset = params["binary_data_offset"].GetInt();
+        rapidjson::Value::ConstMemberIterator iter =
+            params.FindMember("binary_data_size");
+        if (iter != params.MemberEnd()) {
+          byte_size = iter->value.GetInt();
+        } else {
+          const char* dtype = request_input["datatype"].GetString();
+          int element_size = GetDataTypeByteSize(dtype);
+          if (element_size == 0) {
+            return TRTSERVER_ErrorNew(
+                TRTSERVER_ERROR_INVALID_ARG,
+                std::string(
+                    "must specify 'binary_data_size' for input '" +
+                    std::string(request_input["name"].GetString()) +
+                    "' of datatype BYTES")
+                    .c_str());
+          }
+          const rapidjson::Value& shape = request_input["shape"];
+          int element_cnt = 0;
+          for (rapidjson::SizeType i = 0; i < shape.Size(); i++) {
+            if (element_cnt == 0) {
+              element_cnt = shape[i].GetInt();
+            } else {
+              element_cnt *= shape[i].GetInt();
+            }
+          }
+          byte_size = element_cnt * element_size;
+        }
 
-      RETURN_IF_ERR(TRTSERVER_InferenceRequestProviderSetInputData(
-          request_provider, request_input["name"].GetString(), base, byte_size,
-          TRTSERVER_MEMORY_CPU, 0 /* memory_type_id */));
+        int v_idx_tmp = v_idx;
```

Comment:
v_idx_tmp? There is only one v_idx which walks over all the data.

Reply:
This is needed to manage the case where the binary inputs in the request buffer may not in the order of the inputs in the header i.e. INPUT0 could be after INPUT1 in the binary data section after the json header but the INPUT0 may be before INPUT1 in the json section

Reply:
This can be simplified by dropping "offset" so we should do that.

Reply:
Assuming that the offset is removed, we can simply rely on the byte size and will only go through the vecs once